### PR TITLE
DietPi-Software | Replace youtube-dl with yt-dlp

### PR DESCRIPTION
**Pull request by pulpe**
Based on discussion MichaIng/DietPi#6374

Should be ok to merge, tested on virtual dietpi.

### CODE REVIEW

**Comment by MichaIng** [comment]
Many thanks!

File: dietpi/dietpi-software [hunk 1]
```diff
@@ -5647,15 +5647,14 @@ _EOF_
...
 		then
-			aDEPS=('python3')
-			Download_Install 'https://yt-dl.org/downloads/latest/youtube-dl' /usr/local/bin/youtube-dl
-			G_EXEC chmod +x /usr/local/bin/youtube-dl
+			# Remove obsolete youtube-dl and youtube-dl-py2 we used previously
+			G_EXEC rm -f /usr/local/bin/youtube-dl{,-py2}
 
-			# youtube-dl supports Python 2 and Python 3, but its shebang calls Python 2, else fails: https://github.com/ytdl-org/youtube-dl/issues/27649
-			G_EXEC sed -i '1s/python$/python3/' /usr/local/bin/youtube-dl
+			aDEPS=('python3' 'ffmpeg')
```
Comment (MichaIng) [inline dietpi/dietpi-software]: Why is FFmpeg required? If so, then it should be installed as dependency on software ID 7 (which is FFmpeg).

Comment (pulpe) [reply dietpi/dietpi-software]: As stated in [dependencies](https://github.com/yt-dlp/yt-dlp#dependencies). 

> While all the other dependencies are optional, ffmpeg and ffprobe are highly recommended
> 
> Required for [merging separate video and audio files](https://github.com/yt-dlp/yt-dlp#format-selection) as well as for various [post-processing](https://github.com/yt-dlp/yt-dlp#post-processing-options) tasks.

I'll update dependencies then.

File: dietpi/dietpi-software [hunk 2]
```diff
@@ -5647,15 +5648,14 @@ _EOF_
...
 
-			# youtube-dl supports Python 2 and Python 3, but its shebang calls Python 2, else fails: https://github.com/ytdl-org/youtube-dl/issues/27649
-			G_EXEC sed -i '1s/python$/python3/' /usr/local/bin/youtube-dl
+			aDEPS=('python3')
+			Download_Install 'https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp' /usr/local/bin/yt-dlp
+			G_EXEC chmod +x /usr/local/bin/yt-dlp
 
-			# Remove obsolete /usr/local/bin/youtube-dl-py2 we used previously
-			[[ -f '/usr/local/bin/youtube-dl-py2' ]] && G_EXEC rm /usr/local/bin/youtube-dl-py2
+			G_EXEC ln -s /usr/local/bin/yt-dlp /usr/local/bin/youtube-dl
```
Comment (MichaIng) [inline dietpi/dietpi-software]: Skip the removal before the download, so users as not left without any binary if the download fails for some reason. Instead use `-f` option to overwrite any existing file or symlink here:
```suggestion
			G_EXEC ln -sf /usr/local/bin/yt-dlp /usr/local/bin/youtube-dl
```
The py2 removal could/should then stay where it was.

Comment (pulpe) [reply dietpi/dietpi-software]: Done.

**Comment by MichaIng** [comment]
Look good! I'll add changelog and meta stuff and then merge.

What we can have another look at is the other install options:
- Standalone binaries for x86_64, ARMv7 and ARMv8, so no Python 3 required
- Install via pip, makes it a little easier to update: https://pypi.org/project/yt-dlp/
- There is an APT PPA, making it easiest to update via APT: https://launchpad.net/~tomtomtom/+archive/ubuntu/yt-dlp
    But as often, Ubuntu first. Often these can be installed on Debian just fine, but needs testing. It might be problematic if the packages pull specific Python versions as dependency.

**Review by MichaIng — APPROVED** [review]

**Comment by MichaIng** [comment]
Install tests run: https://github.com/MichaIng/DietPi/actions/runs/5007448222

**Pull request merged**